src/services/workspace-client/devWorkspaceClient.ts

```diff
@@ -75,8 +75,13 @@ export class DevWorkspaceClient extends WorkspaceClient {
       await delay();
     }
     this.checkForDevWorkspaceError(workspace);
-    if (!workspace.status || !workspace.status.phase || !workspace.status.ideUrl) {
+    const workspaceStatus = workspace.status;
+    if (!workspaceStatus || !workspaceStatus.phase) {
       throw new Error(`Could not retrieve devworkspace status information from ${workspaceName} in namespace ${namespace}`);
+    } else if (workspaceStatus.phase.toUpperCase() === DevWorkspaceStatus[DevWorkspaceStatus.RUNNING] && !workspaceStatus?.ideUrl) {
+      throw new Error('Could not retrieve ideUrl for the running workspace');
+    } else if (workspaceStatus.message === 'Waiting for workspace routing objects') {
```

Comment:
how much we wait before checking this?

Reply:
It's currently 10 seconds. We can increase it but then we are increasing the time it takes for the factory flow to finish, even on workspaces that wouldn't run into this issue

Reply:
I would vote for removing this check for time being.
Once we propagate the phase to logs, it should not be a big issue.
So, workspace will be in starting phase forever if Che Routing is not installed properly.

Reply:
Do you mean remove the `waiting for workspace routing objects` check? Should I keep the ideUrl check or do you think I should just close this PR?